### PR TITLE
Shell cleanup

### PR DESCRIPTION
**Pull request by bdossantos**
Hello,

I've added some small style improvement to `./diff-so-fancy` shell script according [Shell Style Guide](https://google.github.io/styleguide/shell.xml) and [Shellcheck](https://github.com/koalaman/shellcheck)

### CODE REVIEW

**Comment by stevemao** [comment]
LGTM :)
Can you add Shellcheck to the build scripts?

**Comment by bdossantos** [comment]
Sure, I'll add it in a different PR.

**Comment by bdossantos** [comment]
@stevemao done #64

**Comment by stevemao** [comment]
Nice, Thanks for this :) I really like styling/linting check but I'll see how other collaborators like it.

File: diff-so-fancy [hunk 1]
```diff
@@ -1,15 +1,22 @@
...
-	  src="$(readlink "$src")"
-	  [[ $src != /* ]] && src="$dir/$src"
-	done
-	echo $( cd -P "$( dirname "$src" )" && pwd )
+  src="${BASH_SOURCE[0]}"
+
+  while [[ -h "$src" ]]; do
+    dir="$(cd -P "$( dirname "$src" )" && pwd)"
+    src="$(readlink "$src")"
+    [[ $src != /* ]] && src="$dir/$src"
```
Comment (noscript) [inline diff-so-fancy]: `if..fi`

File: diff-so-fancy [hunk 2]
```diff
@@ -30,32 +37,38 @@ invert_color="\x1B\[7m"
...
 format_diff_header () {
-	# simplify the unified patch diff header
-	$SED -E "s/^($color_code_regex)diff --git .*$//g" | \
-	$SED -E "/$git_index_line_pattern/{N;s/$git_index_line_pattern\n//}" | \
-	$SED -E "s/^($color_code_regex)\-\-\-(.*)$/\1$(print_horizontal_rule)\\
+  # simplify the unified patch diff header
+  $SED -E "s/^($color_code_regex)diff --git .*$//g" | \
+  $SED -E "/$git_index_line_pattern/{N;s/$git_index_line_pattern\n//}" | \
+  $SED -E "s/^($color_code_regex)\-\-\-(.*)$/\1$(print_horizontal_rule)\\
 \1\-\-\-\5/g" | \
```
Comment (noscript) [inline diff-so-fancy]: `$SED -E "s/^($color_code_regex)\-\-\-(.*)$/\1$(print_horizontal_rule)\\n\1\-\-\-\5/g" | \`

File: diff-so-fancy [hunk 3]
```diff
@@ -30,32 +37,38 @@ invert_color="\x1B\[7m"
...
-	$SED -E "s/^($color_code_regex)\+\+\+(.*)$/\1\+\+\+\5\\
+  $SED -E "s/^($color_code_regex)\+\+\+(.*)$/\1\+\+\+\5\\
```
Comment (noscript) [inline diff-so-fancy]: `$SED -E "s/^($color_code_regex)\+\+\+(.*)$/\1\+\+\+\5\\n\1$(print_horizontal_rule)/g"`

**Comment by noscript** [comment]
Overall :+1:

**Comment by bdossantos** [comment]
@noscript I pushed some fixup, I'll rebase if it's OK for you :-)

**Comment by noscript** [comment]
Do we want this squashed? @stevemao @paulirish

**Comment by stevemao** [comment]
Let's wait until Paul's back on this :)

**Comment by paulirish** [comment]
Overall I like a lot of this. 

Can we retain some of the terseness of the top bits ($input and get_script_dir)?  They are fairly unimportant and i'd prefer to not absorb so much space with boring util work.  That said, I def want to stay within shellcheck's bounds, but I'm willing to bend style a tad. What do you think?

**Comment by bdossantos** [comment]
@paulirish Done, I'll squash commits if It's OK for you :)

**Comment by paulirish** [comment]
Yup! squash away. :)

thanks. this is great stuff.

**Comment by paulirish** [comment]
btw is there a tool for enforcing bash style we could adopt?

**Comment by bdossantos** [comment]
Unfortunately shellcheck don't have an "auto fix" feature like some others static code linter (eg: [Rubocop](https://github.com/bbatsov/rubocop)), but I think it's better to run it on Travis on each PR and eventually fix style problems manually.

Travis will be broken if someone doesn't execute Shellcheck locally via a pre-commit hook or directly inside a text editor via a plugin like syntastic (better).

nb: a [.editorconfig](http://editorconfig.org/) file can also help to enforce indentation etc.

**Comment by paulirish** [comment]
gotcha.
yeah true... editorconfig is a good call.

you still squashing?

**Comment by bdossantos** [comment]
It's ok for me, I've also rebased my branch on #64

File: diff-so-fancy (hunk 2 above)
Comment (noscript) [inline diff-so-fancy]: Was merged with no attention to details #72

**Pull request merged**